packages/core/src/components/radio/radio.scss

```diff
@@ -0,0 +1,105 @@
+@use '../../helpers';
+@use '../../mixins';
+
+@mixin Radio() {
+  $transition: 0.15s ease-out;
+  $input-size: helpers.space(2.75);
+  $input-margin: 1px;
+
+  .ods-radio-label {
+    @include helpers.font('300-regular');
+    @include mixins.interactive();
+
+    $padding: (helpers.space(6) - $input-size - $input-margin * 2) / 2;
+
+    --color: #{helpers.color('content-main')};
+
+    color: var(--color);
+    display: inline-flex;
+    gap: $padding;
+    padding: $padding;
+    position: relative;
+
+    &.-bordered {
+      --border-color: #{helpers.color('border-separator')};
+
+      background-color: var(--background-color);
+      border: 1px solid var(--border-color);
+      border-radius: helpers.border-radius('medium');
+      display: flex;
+      transition: $transition;
+      transition-property: background-color;
+    }
+
+    &.-checked {
+      --background-color: #{helpers.color('background-action-subtle')};
+      --border-color: #{helpers.color('border-action-subtle')};
+    }
+
+    &.-disabled {
+      --background-color: unset;
+      --border-color: #{helpers.color('border-disabled')};
+      --color: #{helpers.color('content-disabled')};
+    }
+
+    > :not([aria-hidden='true']) {
```

Comment:
You're right, this indeed applies to `input` also...

I'm not a fan of using `*` as it's slow, however don't see other way.

One could be:

```scss
> span {
  padding-right: helpers.space(2) - $padding;
}
> [aria-hidden='true'] {
  padding: 0;
}
```

But also not great.

Reply:
To be fair, CSS selector optimisation is misunderstood, and the universal rule `*` is one of the most prejudiced against.
According to multiple studies, the performance difference is negligible in most selectors.

https://meiert.com/en/blog/performance-of-css-selectors-2
> What we saw, namely, was an undue focus on selector performance and an outright childish ban of selectors, notably the universal selector.

https://www.stevesouders.com/blog/2009/03/10/performance-impact-of-css-selectors
> For most web sites, the possible performance gains from optimizing CSS selectors will be small, and are not worth the costs.

The focus should be on CSS rules that trigger repaint/reflow or that are just extremely unoptimised.

With that said, however, I don't like using `*` where it can be helped either.

I like your solution, overriding the value on more specific selectors.
However we'd need to include `input` too.

```scss
> input,
> [aria-hidden='true'] {
  padding: 0;
}
```

In which case the `:not` can work just as well
```scss
> :not(input),
> :not([aria-hidden='true']) {
  padding-right: helpers.space(2) - $padding;
}
```

**TL;DR:** I'm happy with the simplest approach which is not changing this selector, and including `padding: 0` on `.ods-radio-input`.

Reply:
Not a fan of `:not` here, would latest change (originally suggested by you) be fine? I've already pushed that change.

Reply:
Yea that's fine. We might update it too if we go with the CSS `:checked` idea and it works.